### PR TITLE
Fix #7996 ⁃ Show Search Suggestions as a list

### LINKED ISSUE #7996: FXIOS-1499 ⁃ Show Search Suggestions as a list
**Vision**

This ticket is to prep for added Append search suggestion (FXIOS-1493), by updating the search suggestions to a list

**Acceptance Criteria**

* Search suggestions appear as a list at the top of the awesomebar results, as shown in mock
  * 4 search suggestions are shown with magnifying glass icon next to them
  * The first item mirrors any character you’ve typed
  * The 2nd, 3rd and 4th items are suggestions we get from the selected search provider
  * EN only: For search suggestion list items, the matching keyword itself is not bolded. The rest of the suggestion string is bolded. For example, if the keyword I typed is “am”, then the suggestion strings may be “am*azon*” or “am*i insurance*”**
  * On both light and dark themes, existing colour values for text and colour fills are used**
* Should be supported for both iPhone and iPad

**Not Included**

* Adding the append arrow is not included in this ticket

**Spec**

* [Mocks](https://www.figma.com/file/lfgj15zbNkL9Pp4hdZGJgp/Proton-Mobile-UI?node-id=51%3A7706)

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1499)

### PR DESCRIPTION
**Pull request by dnarcese**

### CODE REVIEW

File: Client/Frontend/Browser/SearchViewController.swift [hunk 1]
```diff
@@ -367,35 +364,34 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
...
         switch SearchListSection(rawValue: indexPath.section)! {
         case .searchSuggestions:
-            suggestionCell.imageView?.image = searchEngines.defaultEngine.image
-            suggestionCell.imageView?.isAccessibilityElement = true
-            suggestionCell.imageView?.accessibilityLabel = String(format: .SearchSearchEngineSuggestionAccessibilityLabel, searchEngines.defaultEngine.shortName)
-            return suggestionCell
-
+            if let site = suggestions?[indexPath.row] {
+                if let cell = cell as? TwoLineTableViewCell {
+                    if Locale.current.languageCode == "en" {
```
Comment (nbhasin2) [inline Client/Frontend/Browser/SearchViewController.swift:380]: - Wondering if we can cleanup this if let chaining with something like below
`if let site = suggestions?[indexPath.row], let cell = cell as? TwoLineTableViewCell {`

File: Client/Frontend/Browser/SearchViewController.swift [hunk 2]
```diff
@@ -367,35 +364,34 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
...
     }
 
     override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         return 0
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAt: indexPath)
         switch SearchListSection(rawValue: indexPath.section)! {
         case .searchSuggestions:
```
Comment (nbhasin2) [inline Client/Frontend/Browser/SearchViewController.swift:377]: Cases like these can become big that would bloat our `cellForRowAt method` when we add `Sync` or any other cell type. A suggestion would be to add handle method outside `cellForRowAt` that returns appropriate cell

**Comment by isabelrios** [comment]
These tests also check/use the search suggestions, they will need the change from `buttons` to `cells`: https://github.com/mozilla-mobile/firefox-ios/blob/main/XCUITests/SearchTest.swift#L18

**Comment by dnarcese** [comment]
@isabelrios is it strange that they didn't fail here?

**Comment by isabelrios** [comment]
> @isabelrios is it strange that they didn't fail here?

Because the only two that run as part of the Smoke test, are working, don't need changes: 
SearchTests
    ✓ testSearchEngine (77.232 seconds)
    ✓ testSearchStartAfterTypingTwoWords (13.295 seconds)
There are more that run as part of the general test suite, for example, if you run https://github.com/mozilla-mobile/firefox-ios/blob/main/XCUITests/SearchTest.swift#L33, in the same test suite, same issue as we talked before happens. But since the build is green now, the smoketest work we can fix these or other test that may fail in a follow up. I just remembered about this test suite and its tests that are totally related to the PR, that's why I shared.

**Comment by nbhasin2** [comment]
https://user-images.githubusercontent.com/8919439/111697227-b05f8580-880b-11eb-8bd4-d3dc92d00b9e.mov

@dnarcese if you look at first suggestion it becomes bold for a second before turning regular
You can fix it later but bringing it up as I am testing more

**Comment by nbhasin2** [comment]
Another P2 I found is that I see stars beside suggestions while typing ^ see the video above

File: Client/Frontend/Browser/SearchViewController.swift [hunk 3]
```diff
@@ -367,56 +364,26 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
...
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         switch SearchListSection(rawValue: section)! {
         case .searchSuggestions:
-            return searchEngines.shouldShowSearchSuggestions && !searchQuery.looksLikeAURL() && !isPrivate ? 1 : 0
+            if let count = suggestions?.count {
+                return count < 4 ? count : 4
+            } else {
+                return 0
+            }
```
Comment (nbhasin2) [inline Client/Frontend/Browser/SearchViewController.swift:386]: nit: minor cleanup with guard

```suggestion
            guard let count = suggestions?.count else { return 0 }
            return count < 4 ? count : 4
```

**Review by nbhasin2 — APPROVED** [review]

**Comment by nbhasin2** [comment]
Approving these changes so that we can merge and start testing it thoroughly as maybe I missed something with my local testing 😅

**Pull request merged**